pkg/instrumentation/sdk.go

```diff
@@ -299,6 +313,17 @@ func chooseServiceName(pod corev1.Pod, resources map[string]string, index int) s
 	return pod.Spec.Containers[index].Name
 }
 
+// obtains version by splitting image string on ":" and extracting final element from resulting array
+func chooseServiceVersion(pod corev1.Pod, index int) string {
+	parts := strings.Split(pod.Spec.Containers[index].Image, ":")
+	tag := parts[len(parts)-1]
+	//guard statement to handle case where image name has a port number
```

Comment:
is there a test for this guard?

Reply:
I guess i'm not aware of when an image would have a port number? or how checking if `/` is present means there is a port present.

Reply:
Was trying to cover all our bases and noticed the [doc](https://kubernetes.io/docs/concepts/containers/images/#image-names) mentions port numbers as a possibility.

"Images can also include a registry hostname; for example: fictional.registry.example/imagename, and possibly a port number as well; for example: fictional.registry.example:10443/imagename."

I can nix it if you'd like, but if we want to handle that edge case I'm equally happy to write a test for that scenario.
Do you have a preference either way?

Reply:
oh wow i've never seen that before... I think you can keep it in but yeah maybe include a test for it in `sdk-test.go`

Reply:
Added in a test that should do the trick!

Reply:
It would be good to handle the registry ports as well and add more tests:
* with registry port
*  specific tag
*  no tag
*  latest

Reply:
These cases are already handled.

- with registry port is handled here: https://github.com/open-telemetry/opentelemetry-operator/pull/1736/files#diff-260da5a412521a337a3a193ef1e3a344d0e745696b643769367d591a91f84d16R379
- no tag is handled here: https://github.com/open-telemetry/opentelemetry-operator/pull/1736/files#diff-260da5a412521a337a3a193ef1e3a344d0e745696b643769367d591a91f84d16R379
- latest is handled here: https://github.com/open-telemetry/opentelemetry-operator/pull/1736/files#diff-260da5a412521a337a3a193ef1e3a344d0e745696b643769367d591a91f84d16R147 (we actually handle this in TWENTY test cases)

I'm not sure what you mean by 'specific' tag? Can you show me another test in this file that addresses similar concerns?

According to the Kubernetes [docs around image tag naming specifications](https://kubernetes.io/docs/concepts/containers/images/#image-names) these tests already the possible range of values.